dashboard/app/models/pd/workshop.rb

```diff
@@ -224,6 +224,10 @@ def self.end_in_days(days)
     Pd::Workshop.joins(:sessions).group(:pd_workshop_id).having("(DATE(MAX(end)) = ?)", Date.today + days.days)
   end
 
+  def self.should_have_ended
+    self.in_state(STATE_IN_PROGRESS).where("started_at < ?", Time.zone.now - 24.hours)
```

Comment:
This query is not correct. Some workshops are multiple days (sessions). We don't want to enforce workshops only being open for one day, but rather workshops only remaining open one day after the last scheduled session ends. You can leverage `end_on_or_before` for this.

Also note that our servers are all UTC, which is 5-8 hours ahead of the continental US. Workshop session times not adjusted to time zone for simplicity because we don't necessarily know where the workshop will take place. Maybe we should add a buffer so we're not so aggressive, starting in say 36 or 48 hours?